### PR TITLE
feat: Add command "vscode-neovim.lua"

### PR DESCRIPTION
**Pull request by xiyaowong**
improve: Make vscode-neovim.stop can be executed multiple times

1. Customizing the when context to make keybindings more flexible.
2. Executing multiple commands.
3. ...

### CODE REVIEW

**Review by theol0403 — APPROVED** [review]
Can you think of a better place to put the lua handler? Maybe in commands manager or typing manager? It would be nice to have in the same place as nvim.input.

**Comment by xiyaowong** [comment]
I personally think it would be more appropriate to put it in main_controller. commands_manager is not suitable and will be removed in the future.

File: src/main_controller.ts [hunk 1]
```diff
@@ -157,6 +157,17 @@ export class MainController implements vscode.Disposable {
...
             vscode.commands.registerCommand("_getNeovimClient", () => this.client),
+            vscode.commands.registerCommand("vscode-neovim.lua", async (lua) => {
+                if (!lua) {
+                    window.showWarningMessage("No lua code provided");
+                    return;
+                }
+                try {
+                    await this.client.lua(lua);
+                } catch (e) {
+                    logger.error(e instanceof Error ? e.message : e);
```
Comment (justinmk) [inline src/main_controller.ts:168]: We should add a `vscode.commands.registerCommand` wrapper that all commands use, which could handle errors and show/log them.

Comment (xiyaowong) [reply src/main_controller.ts:168]: It's unnecessary. Handling errors or not is the responsibility of the handler.

Comment (justinmk) [reply src/main_controller.ts:168]: Yes, but at the "top level" in most cases the handling is the same: log it and show a message. That avoids lots of redundant try/catch deeper in the codepath.

When code actually has something useful to do with an error, it of course can use a try/catch.

**Pull request merged**